### PR TITLE
Modified for Thread Safety

### PR DESCRIPTION
**Pull request by ErikRBergman**
* Modified TinyPubSub for thread safety.
* .NET Standard version moved from 1.0 to 1.1.

### CODE REVIEW

File: src/TinyPubSub/TinyPubSub.cs [hunk 1]
```diff
@@ -22,57 +22,48 @@
...
-            }
-            else
-            {
-                subscriptions = _channels[channel];
-            }
-            return subscriptions;
+            return _channels.GetOrAdd(channel, key => new ConcurrentDictionary<ISubscription, ISubscription>());
         }
 
+        // EB: Rename to CreateAndAdd?
```
Comment (johankson) [inline src/TinyPubSub/TinyPubSub.cs:43]: Rename ok

File: src/TinyPubSub/TinyPubSub.cs [hunk 2]
```diff
@@ -22,57 +22,48 @@
...
             var current = GetOrCreateChannel(channel);
-            var subscription = new Subscription<string>(owner, action);
-            subscription.RemoveAfterUse = disableAfterFirstUse;
-            current.Add(subscription);
+            var subscription = new Subscription<string>(owner, action, disableAfterFirstUse);
+            current.TryAdd(subscription, subscription);
             return subscription;
         }
 
+        // EB: Rename to CreateAndAdd?
```
Comment (johankson) [inline src/TinyPubSub/TinyPubSub.cs:52]: Rename ok

File: src/TinyPubSub/TinyPubSub.cs [hunk 3]
```diff
@@ -352,10 +343,12 @@ public static void Publish(string channel, string argument = default(string), Ac
         /// <param name="argument">An optional parameter</param>
         /// <remarks>This method is not blocking, it simply uses a Task.Run(() => Publish(...)) internally
         /// to hand of the call to be handled by someone else.</remarks>
-        public static void PublishAsTask(string channel, string argument = default(string), Action<Exception, ISubscription> OnError = null)
+        public static void PublishAsTask(string channel, string argument = default(string), Action<Exception, ISubscription> onError = null)
         {
             // Add to delayed handle queue
-            Task.Run(() => Publish(channel, argument, OnError)).ConfigureAwait(false);
+            //// EB: Configure await is only required when awaiting a task, not creating a fire and forget task
```
Comment (johankson) [inline src/TinyPubSub/TinyPubSub.cs:349]: Thanks for sharing knowledge, did not now that.

File: src/TinyPubSub/TinyPubSub.cs [hunk 4]
```diff
@@ -383,6 +378,8 @@ public static void Clear()
         /// <param name="obj">The object to scan</param>
         public static void Register(object obj)
         {
+            //// TODO: EB: Move the reflection code to a separate type, for performance - add a cache (ConcurrentDictionary) to scanned objects and use expressions to invoke the subscriber methods instead of method.Invoke()..
```
Comment (johankson) [inline src/TinyPubSub/TinyPubSub.cs:381]: Good point, add it as an issue?

**Review by johankson — CHANGES_REQUESTED** [review]
Looks good. Look through the comments if you want to address the issues, otherwise I'll approve this PR.

**Comment by ErikRBergman** [comment]
I guess the "issues" can be addressed later on. The weak reference PR is based on this one, so to avoid merging I propose they are changed later.

**Review by johankson — APPROVED** [review]

**Pull request merged**